gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/RecordListItemViewHolder.java

```diff
@@ -18,74 +18,46 @@
 
 import android.content.Context;
 import android.content.res.Resources;
-import android.text.TextUtils.TruncateAt;
-import android.view.LayoutInflater;
-import android.view.View;
-import android.view.ViewGroup;
-import android.widget.TableRow;
+import android.text.TextUtils;
 import android.widget.TextView;
 import androidx.annotation.NonNull;
-import androidx.lifecycle.MutableLiveData;
 import androidx.recyclerview.widget.RecyclerView;
-import butterknife.BindView;
-import butterknife.ButterKnife;
 import com.google.android.gnd.R;
+import com.google.android.gnd.databinding.RecordListItemBinding;
 import com.google.android.gnd.model.form.Element;
 import com.google.android.gnd.model.form.Field;
 import com.google.android.gnd.model.form.Form;
 import com.google.android.gnd.model.observation.Record;
 import com.google.android.gnd.model.observation.Response;
-import com.google.android.gnd.system.AuthenticationManager;
-import java.text.DateFormat;
-import java.util.Date;
+import com.google.android.gnd.ui.common.Consumer;
 import java8.util.Optional;
 
 class RecordListItemViewHolder extends RecyclerView.ViewHolder {
-  private static final int MAX_SUMMARY_COLUMNS = 4;
-  private final View view;
-  private final MutableLiveData<Record> itemClicks;
-
-  @BindView(R.id.user_name)
-  TextView userNameView;
-
-  @BindView(R.id.last_modified_date)
-  TextView lastModifiedDateView;
 
-  @BindView(R.id.last_modified_time)
-  TextView lastModifiedTimeView;
-
-  @BindView(R.id.field_label_row)
-  TableRow fieldLabelRow;
+  private static final int MAX_SUMMARY_COLUMNS = 4;
 
-  @BindView(R.id.field_value_row)
-  TableRow fieldValueRow;
+  private final Consumer<Record> consumer;
+  private final RecordListItemBinding binding;
 
-  public static RecordListItemViewHolder newInstance(
-      ViewGroup parent, MutableLiveData<Record> itemClicks) {
-    LayoutInflater inflater = LayoutInflater.from(parent.getContext());
-    View view = inflater.inflate(R.layout.record_list_item, parent, false);
-    return new RecordListItemViewHolder(view, itemClicks);
+  RecordListItemViewHolder(
+      @NonNull RecordListItemBinding binding, @NonNull Consumer<Record> consumer) {
+    super(binding.getRoot());
+    this.binding = binding;
+    this.consumer = consumer;
   }
 
-  private RecordListItemViewHolder(View view, MutableLiveData<Record> itemClicks) {
-    super(view);
-    this.view = view;
-    this.itemClicks = itemClicks;
-    ButterKnife.bind(this, view);
-  }
+  public void bind(Record record) {
+    binding.setRecord(record);
+    binding.setConsumer(consumer);
+    binding.executePendingBindings();
 
-  void update(Record record) {
-    updateHeading(record);
-    updatePreview(record);
+    // add label/value for each field
```

Comment:
"Add UI elements for each field with data."

Reply:
Done